internal/appsec/waf/waf_test.go

```diff
@@ -406,6 +407,105 @@ func TestRunError(t *testing.T) {
 	}
 }
 
+func TestMetrics(t *testing.T) {
+	rules := `
+{
+  "version": "2.1",
+  "metadata": {
+    "rules_version": "1.2.7"
+  },
+  "rules": [
+	{
+	  "id": "valid-rule",
+	  "name": "Unicode Full/Half Width Abuse Attack Attempt",
+	  "tags": {
+		"type": "http_protocol_violation"
+	  },
+	  "conditions": [
+		{
+		  "parameters": {
+			"inputs": [
+			  {
+				"address": "server.request.uri.raw"
+			  }
+			],
+			"regex": "\\%u[fF]{2}[0-9a-fA-F]{2}"
+		  },
+		  "operator": "match_regex"
+		}
+	  ],
+	  "transformers": []
+	},
+	{
+	  "id": "missing-tags-1",
+	  "name": "Unicode Full/Half Width Abuse Attack Attempt",
+	  "conditions": [
+	  ],
+	  "transformers": []
+	},
+	{
+	  "id": "missing-tags-2",
+	  "name": "Unicode Full/Half Width Abuse Attack Attempt",
+	  "conditions": [
+	  ],
+	  "transformers": []
+	},
+	{
+	  "id": "missing-name",
+	  "tags": {
+		"type": "http_protocol_violation"
+	  },
+	  "conditions": [
+	  ],
+	  "transformers": []
+	}
+  ]
+}
+`
+	waf, err := NewHandle([]byte(rules))
+	require.NoError(t, err)
+	defer waf.Close()
+	t.Run("RulesetInfo", func(t *testing.T) {
```

Comment:
Amazing test 👏
If it happens to break too many times in the future due to libddwaf changes, we will have to make it more configurable.

Reply:
Adding a TODO for this.